src/webrtc/callEventHandler.ts

```diff
@@ -66,42 +66,48 @@ export class CallEventHandler {
         this.client.removeListener(RoomEvent.Timeline, this.onRoomTimeline);
     }
 
-    private evaluateEventBuffer = async () => {
-        if (this.client.getSyncState() === SyncState.Syncing) {
-            await Promise.all(this.callEventBuffer.map(event => {
-                this.client.decryptEventIfNeeded(event);
-            }));
-
-            const ignoreCallIds = new Set<String>();
-            // inspect the buffer and mark all calls which have been answered
-            // or hung up before passing them to the call event handler.
-            for (const ev of this.callEventBuffer) {
-                if (ev.getType() === EventType.CallAnswer ||
-                        ev.getType() === EventType.CallHangup) {
-                    ignoreCallIds.add(ev.getContent().call_id);
-                }
-            }
-            // now loop through the buffer chronologically and inject them
-            for (const e of this.callEventBuffer) {
-                if (
-                    e.getType() === EventType.CallInvite &&
-                    ignoreCallIds.has(e.getContent().call_id)
-                ) {
-                    // This call has previously been answered or hung up: ignore it
-                    continue;
+    private evaluateEventBuffer = (): void => {
+        // TODO: why are we not handling errors?
+        void (async () => {
+            if (this.client.getSyncState() === SyncState.Syncing) {
+                await Promise.all(this.callEventBuffer.map(event => {
+                    // TODO: as this is inside a Promise.all I think we should be awaiting this?
+                    void this.client.decryptEventIfNeeded(event);
```

Comment:
I think the intention is that this is awaited.

```suggestion
                    return this.client.decryptEventIfNeeded(event);
```